spec/requests/templates_spec.rb

```diff
@@ -28,10 +28,48 @@
   end
 
   context "editing a template" do
-    let!(:template) { FactoryBot.create(:template, :name => 'foo', :description => 'bar') }
+    let!(:template)     { FactoryBot.create(:template, :name => 'foo', :description => 'bar') }
+    let(:zone)          { FactoryBot.create(:zone, :name => "api_zone") }
+    let(:ems)           { FactoryBot.create(:ems_vmware, :zone => zone) }
+    let(:host)          { FactoryBot.create(:host) }
+    let(:template_openstack)       { FactoryBot.create(:template_openstack, :host => host, :ems_id => ems.id, :raw_power_state => "ACTIVE") }
+    let(:template_openstack1)      { FactoryBot.create(:template_openstack, :host => host, :ems_id => ems.id, :raw_power_state => "ACTIVE") }
+    let(:template_openstack2)      { FactoryBot.create(:template_openstack, :host => host, :ems_id => ems.id, :raw_power_state => "ACTIVE") }
```

Comment:
```suggestion
    let(:template_openstack)       { FactoryBot.create(:template_openstack, :host => host, :ems => ems, :raw_power_state => "ACTIVE") }
    let(:template_openstack1)      { FactoryBot.create(:template_openstack, :host => host, :ems => ems, :raw_power_state => "ACTIVE") }
    let(:template_openstack2)      { FactoryBot.create(:template_openstack, :host => host, :ems => ems, :raw_power_state => "ACTIVE") }
```

Reply:
This change caused an error but I have implemented the other changes fine.